### PR TITLE
docs: support for debugging LinkFree in VS Code

### LINKED ISSUE #7438: [OTHER] Support for debugging LinkFree
### What would you like to share?

During my analysis of the LinkFree issue [here](https://github.com/EddieHubCommunity/LinkFree/issues/7065), I noticed that we currently lack any configuration for debugging purposes.

Although debugging locally in VSCode can be a bit challenging, it is still achievable. Unfortunately, I couldn't provide a solution for this matter due to personal laptop issues, as my laptop is broken, and my office laptop has several restrictions in place.

### My Analysis:
During my analysis, I discovered that this issue can be resolved by adding a configuration to the project. Please refer to the screenshot below:
image

For more detailed information, you can visit the following link: [Debugging in Next.js](https://nextjs.org/docs/pages/building-your-application/configuring/debugging)

### Proposed Actions:
1. I kindly request someone to try implementing this solution and, if everything works properly, push it to the repository. Also, consider adding it as a candidate for the `.gitignore` file.
2. It would be beneficial to create documentation that provides a comprehensive guide on debugging for LinkFree.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ArthVhanesa**
## Fixes Issue
Closes #7438 

## Changes proposed
- Add the Next.js debugging support for LinkFree as well as add relevant documentation.
- Added `.vscode/launch.json` as per the docs of Next.js for debugging.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Added a new page of documentation for debugging in VS Code. Attached the pdf of the docs page.
[Debugging in VS Code.pdf](https://github.com/EddieHubCommunity/LinkFree/files/12342966/Debugging.in.VS.Code1.pdf) (Updated)

## Note to reviewers
Kindly go through attached pdf of docs and feel free to suggest changes.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8624"></a>

### CODE REVIEW

**Review by dan-mba — CHANGES_REQUESTED** [review]
I have no clue how you got `.vscode/launch.json` into git as it should be blocked by our `.gitignore`

I am not a fan of adding any settings directly to someone's code editor setup. I use workspace configurations on almost all of the projects I work on, and would be very irritated if it suddenly got overridden.

I think it would be better to either add instructions on creating the `launch.json` with cut / paste or use a different filename `linkfree-launch.json` and instruct users to rename it if they want to use it.

**Comment by ArthVhanesa** [comment]
> I have no clue how you got `.vscode/launch.json` into git as it should be blocked by our `.gitignore`

I added `.vscode` without changing `.gitignore` as I've explained here in this comment https://github.com/EddieHubCommunity/LinkFree/issues/7438#issuecomment-1673569954. It is a one-time change and it won't get updated after this.
As I've mentioned there `.vscode` will not get unnecessary changes when someone does commit, we don't have to worry about that.

> I think it would be better to either add instructions on creating the `launch.json` with cut / paste or use a different filename `linkfree-launch.json` and instruct users to rename it if they want to use it.

Yes, this is a better idea. 

Should I remove the entire `.vscode` folder and add all the instructions in docs or change the file name of `launch.json` and add relevant instructions in docs?

**Comment by dan-mba** [comment]
> I added `.vscode` without changing `.gitignore` as I've explained here in this comment [#7438 (comment)](https://github.com/EddieHubCommunity/LinkFree/issues/7438#issuecomment-1673569954). It is a one-time change and it won't get updated after this. As I've mentioned there `.vscode` will not get unnecessary changes when someone does commit, we don't have to worry about that.
> 

Tricking git to allow you to commit a file is not a good idea.
The correct way would be to configure gitignore with a negation pattern for the single file you want included.

> > I think it would be better to either add instructions on creating the `launch.json` with cut / paste or use a different filename `linkfree-launch.json` and instruct users to rename it if they want to use it.
> 
> Yes, this is a better idea.
> 
> Should I remove the entire `.vscode` folder and add all the instructions in docs or change the file name of `launch.json` and add relevant instructions in docs?

I think the better way is to add all the instructions to the docs as it avoids any chance at messing with someone's vscode config.

**Comment by ArthVhanesa** [comment]
> Tricking git to allow you to commit a file is not a good idea.
> The correct way would be to configure gitignore with a negation pattern for the single file you want included.

Thank you @dan-mba for suggesting best practices.

> I think the better way is to add all the instructions to the docs as it avoids any chance at messing with someone's vscode config.

Okay, I will revert the last changes and add all the instructions in the docs. 👍

**Review by dan-mba — CHANGES_REQUESTED** [review]
Looks good,but the introduction needs to mention that it doesn't work with the `compose.yml` way of running linkfree. Also a mention that this setup requires mongodb to be available, as `compose.yml` runs both linkfree & mongodb, so some users may not realize this is different.

**Comment by ArthVhanesa** [comment]
Updated the documentation. I've attached the screenshot here.
image

File: pages/docs/environments/debugging-in-vscode.mdx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## Debugging LinkFree with VS Code
+This documentation explains how you can debug your LinkFree frontend and backend code using the Visual Studio Code (VS Code) debugger.
+
+<Alert
+  type="warning"
+  message={
+    <>
+      <strong>NOTE: </strong>
+        This debugging setup is for a local development environment and may not work if you are running LinkFree using Docker Compose in your local setup. Additionally, it requires MongoDB to be available and running.
```
Comment (dan-mba) [inline pages/docs/environments/debugging-in-vscode.mdx:14]: I would change `may not work` to `will not work`. The compose file starts the dev server already, so these commands will have no effect.

**Review by dan-mba — CHANGES_REQUESTED** [review]
It looks good to me with one minor change

File: pages/docs/environments/debugging-in-vscode.mdx (hunk 1 above)
Comment (ArthVhanesa) [reply pages/docs/environments/debugging-in-vscode.mdx]: Yes, that's right. Thanks for your attention to detail.
I've changed it now. Please take a look.

**Comment by ArthVhanesa** [comment]
The proposed change is fixed now.

**Review by dan-mba — APPROVED** [review]
Looks good to me

**Review by Sindhuinti — APPROVED** [review]
LGTM 👍🏻

File: pages/docs/environments/debugging-in-vscode.mdx [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import DocsLayout from "@components/layouts/DocsLayout.js";
+import ClipboardCopy from "@components/ClipboardCopy.js";
+import Link from "@components/Link";
+import Alert from "@components/Alert.js";
+
+## Debugging LinkFree with VS Code
```
Comment (eddiejaoude) [inline pages/docs/environments/debugging-in-vscode.mdx:6]:
```suggestion
## Debugging BioDrop with VS Code

```

File: pages/docs/environments/debugging-in-vscode.mdx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+This documentation explains how you can debug your LinkFree frontend and backend code using the Visual Studio Code (VS Code) debugger.
```
Comment (eddiejaoude) [inline pages/docs/environments/debugging-in-vscode.mdx:7]:
```suggestion
This documentation explains how you can debug your BioDrop frontend and backend code using the Visual Studio Code (VS Code) debugger.
```

File: pages/docs/environments/debugging-in-vscode.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+<Alert
+  type="warning"
+  message={
+    <>
+      <strong>NOTE: </strong>
+        This debugging setup is for a local development environment and will not work if you are running LinkFree using Docker Compose in your local setup. Additionally, it requires MongoDB to be available and running.
```
Comment (eddiejaoude) [inline pages/docs/environments/debugging-in-vscode.mdx:14]:
```suggestion
        This debugging setup is for a local development environment and will not work if you are running BioDrop using Docker Compose in your local setup. Additionally, it requires MongoDB to be available and running.
```

File: pages/docs/environments/debugging-in-vscode.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    </>
+  }
+/>
+
+## Environment setup for debugging in VS Code
+
+1. If you don’t already have a `launch.json` file, create one in the `.vscode` directory at the root of the LinkFree project.
```
Comment (eddiejaoude) [inline pages/docs/environments/debugging-in-vscode.mdx:21]:
```suggestion
1. If you don’t already have a `launch.json` file, create one in the `.vscode` directory at the root of the BioDrop project.
```

**Review by eddiejaoude — APPROVED** [review]
Great collaboration everyone 💪

**Pull request merged**